src/Conversion/ONNXToKrnl/Tensor/OneHot.cpp

```diff
@@ -0,0 +1,189 @@
+//===---------------- OneHot.cpp - Lowering OneHot Op -------------------===//
+//
+// This file lowers the ONNX OneHot Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/ONNX/ONNXShapeHelper.hpp"
+
+using namespace mlir;
+
+struct ONNXOneHotOpLowering : public ConversionPattern {
+  ONNXOneHotOpLowering(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXOneHotOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    ONNXOneHotOpAdaptor operandAdaptor(operands);
+    ONNXOneHotOp oneHotOp = llvm::cast<ONNXOneHotOp>(op);
+    auto loc = op->getLoc();
+
+    Value indices = operandAdaptor.indices();
+    Value depth = operandAdaptor.depth();
+    Value values = operandAdaptor.values();
+
+    auto depthShape = depth.getType().cast<MemRefType>().getShape();
+    auto indicesShape = indices.getType().cast<MemRefType>().getShape();
+    int64_t indicesRank = indicesShape.size();
+
+    // Axis is a required attribute and should have default value of -1.
+    int64_t axisValue = oneHotOp.axis();
+    assert(axisValue >= 0 && axisValue <= indicesRank);
+
+    // Insert an allocation and deallocation for the result of this operation.
+    auto memRefOutType = convertToMemRefType(*op->result_type_begin());
+    auto memRefOutShape = memRefOutType.getShape();
+    int64_t outRank = memRefOutType.getRank();
+
+    IndexExprScope ieScope(rewriter, loc);
+    DimsExpr outputDims(outRank);
+    MemRefBoundsIndexCapture indicesBounds(indices);
+
+    // Allocate result.
+    Value alloc;
+    Value zero = emitConstantOp(rewriter, loc, rewriter.getIndexType(), 0);
+    Value one = emitConstantOp(rewriter, loc, rewriter.getIndexType(), 1);
+
+    // Load values depending on shape.
+    Value loadedDepth;
+    if (depthShape.size() == 0)
+      loadedDepth = rewriter.create<KrnlLoadOp>(loc, depth, ArrayRef<Value>{});
+    else if (depthShape.size() == 1 && depthShape[0] == 1)
```

Comment:
When depth is a scalar as specified, depthShape.size() should be 0. Some previous code may not be correct.

Reply:
OK. So I remove `depthShape.size() == 1 && depthShape[0] == 1` condition. I copy this code from Range. I follow the rule to accept both scalar value or a tensor of length 1.